src/main/scala/org/broadinstitute/dsde/agora/server/ga4gh/ModelSupport.scala

```diff
@@ -0,0 +1,120 @@
+    assert(entityType.nonEmpty, "cannot create a ToolClass if entity type is empty")
+    val str = entityType.get.toString
+    new ToolClass(str, str, "")
+  }
+
+  def toolFromEntities(entities:Seq[AgoraEntity]): Tool = {
+    val representative = entities.last
+    val versions = entities.toList map (x => ModelSupport.toolVersionFromEntity(x))
+    val latestVersion = versions.last
+    val id = ToolId(representative).toString
+    val url = AgoraConfig.GA4GH.toolUrl(id, latestVersion.id, latestVersion.`descriptor-type`.last)
+    val author = findAuthorsInWdl(representative.payload)
+    new Tool(
+      url=url,
+      id=id,
+      organization=organization,
+      toolname=latestVersion.name,
+      toolclass=ToolClass(representative),
+      description=representative.synopsis.getOrElse(""),
+      author=author,
+      `meta-version` = latestVersion.`meta-version`,
+      contains=List.empty[String],
+      verified=false,
+      `verified-source`= verifiedSource,
+      signed=false,
+      versions=versions
+    )
+  }
+
+  def toolVersionFromEntity(entity: AgoraEntity): ToolVersion = {
+    assert(entity.entityType.contains(AgoraEntityType.Workflow), "cannot create a ToolVersion if entityType is not Workflow")
+    assert(entity.snapshotId.nonEmpty, "cannot create a ToolVersion if snapshot id is empty")
+    new ToolVersion(
+      name = entity.name.getOrElse(""),
+      url = entity.url.getOrElse(""),
+      id = ToolId(entity).toString,
+      image = "",
+      `descriptor-type` = List("WDL"),
+      dockerfile = false,
+      `meta-version` = entity.snapshotId.getOrElse(Int.MinValue).toString,
+      verified = false,
+      `verified-source` = verifiedSource
+    )
+  }
+
+  def toolDescriptorFromEntity(entity: AgoraEntity): ToolDescriptor = {
+    new ToolDescriptor(
+      `type` = ToolDescriptorType.WDL,
+      descriptor = entity.payload.getOrElse(""),
+      url = entity.url.getOrElse("")
+    )
+  }
+
+  /**
+    * Metadata is always a constant.
+    */
+  def metadata(): Metadata = {
+    new Metadata(version = version, `api-version` = apiVersion, country = country, `friendly-name` = friendlyName)
+  }
+
+  /**
+   * Looks for all populated "meta: author=X" and "meta: email=Y" fields in the optional wdl meta fields.
+   */
+  def findAuthorsInWdl(payload: Option[String]): String = {
+    val authorField = "author"
+    val emailField = "email"
+    val stringFormat = "%s <%s>"
+    payload match {
+      case Some(wdl) =>
+        WdlNamespaceWithWorkflow.load(wdl, Seq.empty) match {
+          case Success(parsed) =>
+            val authorEmails: Seq[(String, String)] = {
+              parsed.tasks.map(_.meta.getOrElse(authorField, "")).zip(parsed.tasks.map(_.meta.getOrElse(emailField, ""))) ++
+                parsed.workflows.map(_.meta.getOrElse(authorField, "")).zip(parsed.workflows.map(_.meta.getOrElse(emailField, "")))
+            }.filterNot(_._1.isEmpty).distinct
```

Comment:
I love this approach of pulling stuff out of the WDL.

If I read this correctly, the logic is:
* if WDL developer specified author and email, use "author <email>"
* if WDL developer specified only author, use "author"
* if WDL developer specified only email, ignore.

We could handle the last case with code something like:
```
parsed.tasks.map { task =>
                val email = task.meta.getOrElse(emailField,"").trim
                val author = task.meta.getOrElse(authorField,"").trim
                val coalescedAuthor = if (author.nonEmpty) author else email
                author -> email
              }
```

Reply:
You read that correctly, and I like that additional case ... wish I had thought of it :-)

Reply:
The whole thing could be much simpler by handling the string interpolation inside the map as well.